awx/api/generics.py

```diff
@@ -269,12 +270,19 @@ def get_serializer(self, *args, **kwargs):
         return serializer
 
     def get_queryset(self):
-        #if hasattr(self.request.user, 'get_queryset'):
-        #    return self.request.user.get_queryset(self.model)
         if self.queryset is not None:
             return self.queryset._clone()
         elif self.model is not None:
-            return self.model._default_manager.all()
+            qs = self.model._default_manager
+            try:
+                access_class = access_registry[self.model]
+                if access_class.select_related:
+                    qs = qs.select_related(*access_class.select_related)
+                if access_class.prefetch_related:
+                    qs = qs.prefetch_related(*access_class.prefetch_related)
+            except KeyError:
```

Comment:
`KeyError` is a really broad thing to catch, especially for the number of lines in this `try` block.

Could we accomplished the same thing with:

```python
if self.model in access_registry:
    access_class = access_registry[self.model]
    qs = ...
return qs
```

Reply:
This is a good point.